fixture-model-registry/generate.sh

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+while read var; do
+  [ -z "${!var+x}" ] && { echo "$var is empty or not set"; exit 1; }
+done << EOF
+AZURE_STORAGE_CONNECTION_STRING
+EOF
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+USER_NAME="iterative"
+REPO_NAME="fixture-model-registry"
+
+BUILD_PATH="$HERE/build"
+
+mkdir -p $BUILD_PATH
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install -r $HERE/code/requirements.txt
+  pip install -e ~/Git/iterative/gto/
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=30
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE=${TAG_TIME}
+export GIT_COMMITTER_DATE=${TAG_TIME}
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE=${TAG_TIME}
+  export GIT_COMMITTER_DATE=${TAG_TIME}
+}
+
+export GIT_AUTHOR_NAME="Alexander Guschin"
+export GIT_AUTHOR_EMAIL="1aguschin@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+
+git init -b main
+cp -r $HERE/code/ .
+pip freeze > requirements.txt
+git add .
+tick
+git commit -m "Initialize Git repository"
+
+
+echo "Create new models"
+dvc init
+mlem init
+tick
+git add .mlem .dvc
+git commit -m "Initialize MLEM and DVC project"
+
+
+python train.py
```

Comment:
Q: do we need actual code / train or we can imitate it for the fixture?

Reply:
Let's keep it - it create some diversity in trained models (we have some "dummy ML models" saved already). Plus the fact it's a real MLEM model can be used in tests I think.